### PR TITLE
Add version output

### PR DESCRIPTION
**Pull request by radeksimko**
This is likely to conflict with https://github.com/hashicorp/terraform-json/pull/23 - I can rebase the other one when either of these PRs are merged.

--- 

It is planned to parse the JSON output in `terraform-exec` at some point as per https://github.com/hashicorp/terraform-exec/issues/13 and this PR would therefore support these efforts.

The struct is practically copy&paste of the one used in core:

https://github.com/hashicorp/terraform/blob/ae025248cc0712bf53c675dc2fe77af4276dd5cc/command/version.go#L26-L32

I just added `omitempty` to `platform` since that was introduced in v0.15, although I don't expect anyone will actually use these structs for marshalling.

### CODE REVIEW

**Comment by vancluever** [comment]
@radeksimko I've merged your other two PRs, wanna fix this one so that it's ready to be merged?

PS: as there's possible talk about integrating this with some other core repos, I'm going to just hold off on tagging a new version for now unless it's holding up work. I'll leave it to you though to see what you want to do. Let me know!

**Comment by radeksimko** [comment]
@vancluever Rebased. 

> as there's possible talk about integrating this with some other core repos, I'm going to just hold off on tagging a new version for now unless it's holding up work. I'll leave it to you though to see what you want to do. Let me know!

I am aware. We've been briefly chatting about the transition and I think we'd prefer to still cut at least one more release (0.7.0 presumably), so we can use it in https://github.com/hashicorp/terraform-exec/pull/68 (just for consistency really) which in turn unblocks https://github.com/hashicorp/terraform-ls/pull/323

**Comment by vancluever** [comment]
Sounds good. I'll tag shortly!

**Review by vancluever — APPROVED** [review]

**Pull request merged**
